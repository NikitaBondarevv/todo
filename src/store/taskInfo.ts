import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import { IInfo } from 'interfaces/IInfo'
import { TaskInfoState } from 'interfaces/ITaskInfoState'

const initialState: TaskInfoState = {
  data: {
    done: 0,
    inProgress: 0,
    total: 0,
    waiting: 0
  }
}

export const taskInfo = createSlice({
  name: 'taskInfo',
  initialState,
  reducers: {
    setInfo: (state: TaskInfoState, action: PayloadAction<IInfo>) => {
      state.data = action.payload
    }
  },
})

export const { setInfo } = taskInfo.actions

export default taskInfo.reducer
