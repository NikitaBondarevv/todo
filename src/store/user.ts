import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

import { IUser } from 'interfaces/IUser'

export interface UserState {
  data?: IUser
  isAuthenticated: boolean
}

const initialState: UserState = {
  isAuthenticated: false,
}

export const user = createSlice({
  name: 'user',
  initialState,
  reducers: {
    login: (state: UserState, action: PayloadAction<IUser>) => {
      state.data = action.payload
      state.isAuthenticated = true
    },
    
    logout: (state: UserState) => {
      delete state.data
      state.isAuthenticated = false
    },
  },
})

// Action creators are generated for each case reducer function
export const { login, logout } = user.actions

export default user.reducer
