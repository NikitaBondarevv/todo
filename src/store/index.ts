import { configureStore } from '@reduxjs/toolkit'
import { useDispatch, useSelector } from 'react-redux'
import type { TypedUseSelectorHook } from 'react-redux'

import user from './user'
import taskInfo from './taskInfo'

export const store = configureStore({
  reducer: {
    user: user,
    taskInfo: taskInfo
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
