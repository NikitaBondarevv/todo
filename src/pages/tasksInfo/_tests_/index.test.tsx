import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

import { TasksInfo } from '..'
import { store } from '../../../store'

jest.mock('contracts/getInfo', () => ({
  getInfo: () => ({})
}))

describe('<TasksInfo />', () => {
  test('should render div', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Provider store={store}>
          <TasksInfo />
        </Provider>
      </BrowserRouter>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
