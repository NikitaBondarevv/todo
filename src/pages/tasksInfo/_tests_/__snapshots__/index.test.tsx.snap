// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksInfo /> should render div 1`] = `
<DocumentFragment>
  <div
    class="tasksInfo"
  >
    <span>
      Hello, 
    </span>
    <ul
      class="list"
    >
      <li
        class=""
      >
        You have 
        <mark>
          0
        </mark>
         tasks
      </li>
      <li
        class="done"
      >
        Done: 
        <mark>
          0
        </mark>
      </li>
      <li
        class="inProgress"
      >
        In progress: 
        <mark>
          0
        </mark>
      </li>
      <li
        class="waiting"
      >
        Waiting: 
        <mark>
          0
        </mark>
      </li>
    </ul>
    <a
      class="link"
      href="/tasks"
    >
      Go to the task list
    </a>
  </div>
</DocumentFragment>
`;
